src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/Encoding/BlobEncoders.cs

```diff
@@ -608,9 +576,8 @@ public NameEncoder(BlobBuilder builder)
             Builder = builder;
         }
 
-        public void Name(string name)
+        public void Name(string name!!)
         {
-            if (name == null) Throw.ArgumentNull(nameof(name));
             if (name.Length == 0) Throw.ArgumentEmptyString(nameof(name));
 
```

Comment:
I'm a little surprised this is using !! and a length check instead of ArgumentException.ThrowIfNullOrEmpty.  Perhaps the exception message is too different for the empty case?

Reply:
netstandard2.0 :(